Slim/Control/Queries.pm

```diff
@@ -6331,11 +6350,12 @@ sub _getTagDataForTracks {
 	# Bind selected columns in order
 	my $i = 1;
 	for my $col ( @cols ) {
-		# Adjust column names that are sub-queries to be stored using the AS value
-		if ( $col =~ /SELECT/ ) {
-			my ($newcol) = $col =~ /AS (\w+)/;
-			$c->{$newcol} = 1;
-			$col = $newcol;
+		# use custom column name if provided
+		if ( ref $c->{$col} eq 'ARRAY' ) {
+			my $oldcol = $col;
+			$col = $c->{$col}[1];
+			$c->{$col} = 1;
+			delete $c->{$oldcol};
 		}
```

Comment:
That's identical to the above, isn't it? Then let's put it in a helper function. As `$c` is a hash ref, passing it around will automatically update the actual value in the caller, too. 

It's this kind of obscure code which shouldn't be duplicated, as a year or two down the road nobody would remember why we did something the way we did it, and would certainly only change one occurrence of this.

Reply:
I see your point. What about keeping `$c` completely clean by introducing a second map with the same key, optionally created if we want to override the column name?

I think this would also be safer than reverting to parsing the key, because almost anything _could_ be a valid SQL column name as long as it was enclosed in quotes. Demonstrated by the fact that SQL would actually use the whole subselect statement as the column heading, if we didn't give it an `AS`.

On the duplicate block, yes I thought you'd say that, I'd have done it already but I wasn't sure whether to put the whole loop in a function, or just the `if...` part.

Thoughts?

Reply:
> I think this would also be safer than reverting to parsing the key, because almost anything could be a valid SQL column name as long as it was enclosed in quotes. 

What I'm saying is NOT parsing the key, but only to check whether it was part of an SQL query (or fwiw: no valid column name, eg. a space). The actual column name would be the value to the key.

As for the loop vs. only the special handling: I would have opted for the latter. I didn't even look at the rest of the loop. I really thought about encapsulating that relatively odd and particular code in a function with a self explanatory name.

Reply:
But you can have a space in a column name, as long as it's quoted (I know we don't, at present, but still...)

Anyway, I'm about to push some changes, as some source code is better than a thousand words ;)

Reply:
No one in their right mind would use a space in a column name. If they did anyway and got it merged to LMS, they shall be punished by this code!